### DISCUSSION TITLE
betterDefine如何启用?

### DISCUSSION THREAD

**Discussion by MingRock** [vue-macros/vue-macros#148] [General]
在vite配置文件中如何启用,文档中没有找到

**Comment by sxzz** [comment]
所有特性默认开启。配置好 [Get Started](https://vue-macros.sxzz.moe/guide/getting-started.html) 就可以了

**Reply by MingRock** [reply]
modelValue不支持定义在外部d.ts文件中么?
 [Vue warn]: Property "modelValue" was accessed during render but is not defined on instance.)

`

- props.d.ts

export declare interface IOtherProps{
  modelValue: string | number;
}

- component.vue

import type {IOtherProps} from "./props";

interface IProps extends IOtherProps{}
const props = withDefaults(defineProps<IProps>(), {});

`

**Reply by sxzz** [reply]
暂不支持，目前 better-define 仅支持 props 与 emits。defineModel 即将在几天后支持。

**Reply by sxzz** [reply]
你这个 defineProps 应该是支持的，请附上完整复现代码。

**Reply by MingRock** [reply]
> 你这个 defineProps 应该是支持的，请附上完整复现代码。

https://stackblitz.com/edit/vitejs-vite-yegint?file=src%2Fcomponents%2FInputText.vue

**Reply by sxzz** [reply]
```diff
--- import type { InputTextProps } from './types';
+++ import type { InputTextProps } from './types.d';
```

**Reply by MingRock** [reply]
> ```diff
> --- import type { InputTextProps } from './types';
> +++ import type { InputTextProps } from './types.d';
> ```

改完为什么还是不行呢?

**Reply by sxzz** [reply]
<img width="1104" alt="image" src="https://user-images.githubusercontent.com/6481596/202341643-fbe0f02e-1de8-41a2-9b95-bdc29eb7b367.png">
我打开了你 stackblitz.com 的链接，看到应该是正常了

**Reply by MingRock** [reply]
> <img alt="image" width="1104" src="https://user-images.githubusercontent.com/6481596/202341643-fbe0f02e-1de8-41a2-9b95-bdc29eb7b367.png">
> 
> 我打开了你 stackblitz.com 的链接，看到应该是正常了

stackblitz中的最小环境是没问题了,我的项目中,还是有问题,我再检查下吧,谢谢

**Reply by MingRock** [reply]
> <img alt="image" width="1104" src="https://user-images.githubusercontent.com/6481596/202341643-fbe0f02e-1de8-41a2-9b95-bdc29eb7b367.png">
> 
> 我打开了你 stackblitz.com 的链接，看到应该是正常了

还一个问题请教,使用@quasar/vite-plugin插件,会判断是否包含vite:vue名称的插件,使用VueMacros后会使插件列表变为二维数组,导致无法运行
https://stackblitz.com/edit/vitejs-vite-yegint?file=vite.config.ts

vite配置文件改为如下配置就可以了:
defineConfig({
    plugins: [
        VueMacros(),
        vue(),
        quasar(),
    ],
});


请问:
plugins: [VueMacros(),vue()]
与
plugins: [VueMacros({plugins: {vue: vue()}})]
有区别吗?

**Reply by sxzz** [reply]
我认为这是 Quasar 的 Bug。你可以通过
```
    ...VueMacros({
      plugins: {
        vue: Vue(),
      },
    }),
```
暂时规避这个问题。

**Reply by MingRock** [reply]
> 我认为这是 Quasar 的 Bug。你可以通过
> 
> ```
>     ...VueMacros({
>       plugins: {
>         vue: Vue(),
>       },
>     }),
> ```
> 
> 暂时规避这个问题。

可以了,修改import导入的接口内容,不支持hmr update吗?现在必须重启vite才能更新

**Reply by sxzz** [reply]
麻烦提供下复现

**Reply by MingRock** [reply]
> 麻烦提供下复现

https://stackblitz.com/edit/vitejs-vite-yegint?file=src%2Fcomponents%2Ftypes.d.ts

types.d.ts 文件中删除
`test?:string`
属性,或添加属性不触发hmr update,
只有在InputText.vue中的props修改才触发

**Reply by sxzz** [reply]
#153 HMR 暂时还没处理
